CMakeLists.txt

```diff
@@ -78,8 +78,16 @@ option(BUILD_NETOPEER2_LIB "Build netopeer2 library with sysrepo YANG module set
 option(NETOPEER2_LIB_SERVER "Include server main in netopeer2 library" ON)
 option(NETOPEER2_LIB_TESTS "Include server tests in netopeer2 library" ON)
 set(SSH_AUTHORIZED_KEYS_FORMAT "%h/.ssh/authorized_keys" CACHE STRING "sshd-like pattern (with '%h', '%u', '%U') for determining path to users' SSH authorized_keys file.")
-set(THREAD_COUNT 3 CACHE STRING "Number of threads accepting new sessions and handling requests")
-set(POLL_IO_TIMEOUT 10 CACHE STRING "Timeout in milliseconds of polling sessions for new data. It is also used for synchronization of low level IO such as sending a reply while a notification is being sent")
+if(NOT THREAD_COUNT)
+    set(THREAD_COUNT 3 CACHE STRING "Number of threads accepting new sessions and handling requests")
+endif()
+if(NOT POLL_IO_TIMEOUT)
+    set(POLL_IO_TIMEOUT 10 CACHE STRING "Timeout in milliseconds of polling sessions for new data. It is also used for synchronization of low level IO such as sending a reply while a notification is being sent")
+endif()
```

Comment:
I do not see the reason for these changes, cached strings can be changed during compilation and `set` is used only as the default.